pkg/cosign/remote.go

```diff
@@ -95,29 +105,53 @@ func findDuplicate(ctx context.Context, sigImage v1.Image, payload []byte, dupeD
 	return nil, nil
 }
 
-func Upload(ctx context.Context, signature, payload []byte, dst name.Reference, cert, chain string, dupeDetector signature.Verifier, auth authn.Keychain) (uploadedSig []byte, err error) {
+type Bundle struct {
+	SignedEntryTimestamp strfmt.Base64
+	CanonicalizedPayload []byte
+}
+
+type UploadOpts struct {
```

Comment:
I like moving to the opts but don't want it to become just a huge grab bag, it also gets weird for mandatory vs. optional ones. What about separating out the required ones and the optional ones?

Leave signature, payload and dst as actual args, and move cert, chain, dupeDetector, auth and bundle to the struct?

Reply:
good call! I left auth in as well since I think we need it